### PR TITLE
Do not write the conf file on windows

### PR DESCRIPTION
**Pull request by strseb**
## Description

   [ As per #2910 
](https://github.com/mozilla-mobile/mozilla-vpn-client/issues/2910) - We're not writing the config to disk anymore, but passing it directly into the windowstunneldaemon
## Reference

[    i.e Jira or Github issue URL](https://github.com/mozilla-mobile/mozilla-vpn-client/issues/2910)

## Checklist
    
- [x] My code follows the style guidelines for this project
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have performed a self review of my own code
- [x] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

### CODE REVIEW

File: src/wgquickprocess.cpp [hunk 1]
```diff
@@ -24,8 +24,61 @@ Logger logger(
 bool WgQuickProcess::createConfigFile(const QString& outputFile,
```
Comment (bakulf) [inline src/wgquickprocess.cpp:24]: we can remove this method, I guess.

File: src/wgquickprocess.cpp [hunk 2]
```diff
@@ -24,8 +24,61 @@ Logger logger(
...
+    qWarning("Unable to write the whole configuration file");
+    return false;
+  }
+
+  file.close();
+  return true;
+}
+
+// static
+bool WgQuickProcess::createConfigFile(
```
Comment (bakulf) [inline src/wgquickprocess.cpp:49]: do we need this?

**Review by bakulf — APPROVED** [review]

File: src/platforms/windows/daemon/wireguardutilswindows.cpp [hunk 3]
```diff
@@ -80,32 +80,28 @@ QList<WireguardUtils::PeerStatus> WireguardUtilsWindows::getPeerStatus() {
...
-  if (!m_tunnel.start(tunnelFile)) {
+  if (!m_tunnel.start(configString.toString())) {
     logger.error() << "Failed to activate the tunnel service";
     return false;
   }
 
   // Determine the interface LUID
   NET_LUID luid;
-  QString ifAlias = QFileInfo(tunnelFile).baseName();
+  QString ifAlias = "MozillaVPN";
```
Comment (oskirby) [inline src/platforms/windows/daemon/wireguardutilswindows.cpp:104]: You can get this from `WireguardUtils::interfaceName()` which should reduce the number of hard-coded strings floating around.

File: src/wgquickprocess.cpp [hunk 4]
```diff
@@ -21,11 +21,10 @@ Logger logger(
 }  // namespace
 
 // static
-bool WgQuickProcess::createConfigFile(const QString& outputFile,
-                                      const InterfaceConfig& config,
-                                      const QMap<QString, QString>& extra) {
+QVariant WgQuickProcess::createConfigString(
```
Comment (oskirby) [inline src/wgquickprocess.cpp:24]: The use of QVariant here is kind-of unnecessary, we can check for a failed config by testing `QString::isNull()` or `QString::isEmpty()`

File: windows/tunnel/service.go [hunk 5]
```diff
@@ -109,17 +109,13 @@ func (service *tunnelService) Execute(args []string, r <-chan svc.ChangeRequest,
 		return
 	}
 
-	config, err = conf.LoadFromPath(service.Path)
+	config, err = conf.FromWgQuickWithUnknownEncoding(service.ConfString,"MozillaVPN")
```
Comment (oskirby) [inline windows/tunnel/service.go:112]: It feels incorrect to hard-code the tunnel interface name here, this is something that should be passed in by the daemon as a runtime parameter.

**Review by oskirby — APPROVED** [review]

**Pull request merged**